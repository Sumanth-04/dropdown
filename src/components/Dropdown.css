button{ 
    background-color: white;
    border: 1px solid #ccc;
    color: black;
    padding:5px;
    padding-left:8px;
    padding-right:8px;
    text-align: center;
    display: inline-block;
    font-size: 18px;
    border-radius: 3px;
}
button:hover{
    background-color: #ebfaff;
}