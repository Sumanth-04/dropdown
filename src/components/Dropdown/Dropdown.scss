@import '../variables';


.rd-dropdown{ 
    background-color: white;
    border: 1px solid $primary-color ;
    color: $primary-color ;
    padding: 4px 18px;
    text-align: center;
    display: inline-block;
    font-size: 14px;
    margin-left: 12px;
    &:hover{
        background-color: #ebfaff;
    }
}
