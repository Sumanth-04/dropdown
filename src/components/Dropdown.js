import React, { Component } from "react";
import {default as DropdownRc} from "rc-dropdown";
import PropTypes from "prop-types";
import "rc-dropdown/assets/index.css";
import "./Dropdown.css";
class Dropdown extends Component {
  constructor(){
    super();
    this.onClick = this.onClick.bind(this);
    this.onBlur = this.onBlur.bind(this);
    this.state = {
      toggle : false,
    }
  }
  onBlur(){
    setTimeout(()=>{ 
      this.setState({toggle:false});;
    }, 200)
  }
  onClick(){
    this.setState({toggle:!this.state.toggle});
  }
  
  render() {
    return (
      <div>
        <DropdownRc
          className={this.props.className}
          name={this.props.name}
          prefixCls={this.props.prefixCls}
          visible={this.state.toggle}
          defaultVisible={this.props.defaultVisible}
          overlayClassName={this.props.overlayClassName}
          overlay={this.props.children}
          onOverlayClick={this.props.onOverlayClick}
          transitionName={this.props.transitionName}
          animation={this.props.animation}
          onVisibleChange={this.props.onVisibleChange}
          minOverlayWidthMatchTrigger={this.props.minOverlayWidthMatchTrigger}
          getPopupContainer={this.props.getPopupContainer}
          onClick={this.onClick}
          onBlur={this.onBlur}
        >
          <button>
            {this.props.name ? this.props.name : this.props.className}
            {this.props.showCaret? <i className="fa fa-caret-down" aria-hidden="true"></i> : undefined}
          </button>
        </DropdownRc>
      </div>
    );
  }
}

Dropdown.propTypes = {
  className: PropTypes.string.isRequired,
  name: PropTypes.string,
  prefixCls: PropTypes.string,
  showCaret: PropTypes.bool.isRequired,
  transitionName: PropTypes.string,
  animation: PropTypes.string,
  onVisibleChange: PropTypes.func,
  defaultVisible: PropTypes.bool,
  overlay: PropTypes.node,
  onOverlayClick: PropTypes.func,
  minOverlayWidthMatchTrigger: PropTypes.bool,
  getPopupContainer: PropTypes.func
};
Dropdown.defaultProps = {
  className: "Dropdown",
  showCaret: true
};

export default Dropdown;
