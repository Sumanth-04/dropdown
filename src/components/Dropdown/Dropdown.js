import React, { Component } from "react";
import ReactDOM from 'react-dom';
import { default as DropdownRc } from "rc-dropdown";
import PropTypes, { array, node } from "prop-types";
import Menu, { Item as MenuItem } from "rc-menu";
import "./index.scss";
import "./Dropdown.scss";

class Dropdown extends Component {
  constructor(props) {
    super();
    this.onClick = this.onClick.bind(this);
    this.onBlur = this.onBlur.bind(this);
    this.onMenuItemClick = this.onMenuItemClick.bind(this);
    this.state = {
      toggle: false,
      name: props.name,
      className: props.className,
      selected: undefined,
    };
  }
  onBlur() {
    setTimeout(() => {
      this.setState({ toggle: false });
    }, 300);
  }
  onClick(e) {
    e.preventDefault();
    this.setState({ toggle: !this.state.toggle });
  }
  onMenuItemClick(e) {
    this.setState({ selected: e.item.props.children });
  }
  /*componentDidMount(){
    var btnwidth = ReactDOM.findDOMNode(this).getElementsByClassName('rd-dropdown');
    console.log(btnwidth)
  }*/

  render() {
    var arrOfChildren = [];
    var Child ;
    if(this.props.overlay){
      if(typeof(this.props.overlay) != node)
      {  
        arrOfChildren = this.props.overlay.props.children.map((val, index) => {
          return (
            <MenuItem onClick={this.onMenuItemClick} key={index}>
              {val.props.children}
            </MenuItem>
          );
        });
      }else{
        Child = <MenuItem onClick={this.onMenuItemClick}>{this.props.overlay}</MenuItem>
      }
    }else{
      if(typeof(this.props.children) != node)
      {
        arrOfChildren = this.props.children.props.children.map((val, index) => {
          return (
            <MenuItem onClick={this.onMenuItemClick} key={index}>
              {val.props.children}
            </MenuItem>
          );
        });
      }else{
        Child = <MenuItem onClick={this.onMenuItemClick}>{this.props.children}</MenuItem>
      }
    }
    
    var itemstoshow = (
      <Menu onSelect={this.props.onSelect} /*style={{width : this.state.btn_width }}*/>
        {arrOfChildren.map((val, i) => {
          return val;
        })}
      </Menu>
    );

    return (
      <div>
        <DropdownRc
          className={`rd-${this.props.className}`}
          name={this.props.name}
          prefixCls={this.props.prefixCls}
          visible={this.state.toggle}
          defaultVisible={this.props.defaultVisible}
          overlayClassName={this.props.overlayClassName}
          overlay={itemstoshow}
          onOverlayClick={this.props.onOverlayClick}
          transitionName={this.props.transitionName}
          animation={this.props.animation}
          onVisibleChange={this.props.onVisibleChange}
          minOverlayWidthMatchTrigger={this.props.minOverlayWidthMatchTrigger}
          getPopupContainer={this.props.getPopupContainer}
          onClick={this.onClick}
          onBlur={this.onBlur}
          {...this.props}
        >
          <button className="rd-dropdown" style={this.props.style}>
            {this.state.selected === undefined || this.state.selected === ""
              ? this.state.name
                ? this.state.name
                : this.state.className
              : this.state.selected}

            <span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>
            {this.props.showCaret ? (
              <i className="fa fa-caret-down" aria-hidden="true"></i>
            ) : undefined}
          </button>
        </DropdownRc>
      </div>
    );
  }
}

Dropdown.propTypes = {
  onSelect : PropTypes.func,
  className: PropTypes.string.isRequired,
  name: PropTypes.string,
  prefixCls: PropTypes.string,
  showCaret: PropTypes.bool.isRequired,
  transitionName: PropTypes.string,
  animation: PropTypes.string,
  onVisibleChange: PropTypes.func,
  defaultVisible: PropTypes.bool,
  overlay: PropTypes.node,
  onOverlayClick: PropTypes.func,
  minOverlayWidthMatchTrigger: PropTypes.bool,
  getPopupContainer: PropTypes.func,
};
Dropdown.defaultProps = {
  className: "--select--",
  showCaret: true,
};

export default Dropdown;
