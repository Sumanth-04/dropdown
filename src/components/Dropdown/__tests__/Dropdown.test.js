import React from "react";
import Dropdown from "../Dropdown";
import renderer from "react-test-renderer";

it("Dropdown should have a single child and options in that child (Use div tag)", () => {
  const tree = renderer
    .create(
      <Dropdown name="Click me">
        <div>
          <div>Option 1</div>
          <div>Option 2</div>
          <div>Option 3</div>
        </div>
      </Dropdown>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});