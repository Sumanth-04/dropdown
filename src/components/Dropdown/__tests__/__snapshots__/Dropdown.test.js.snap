// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should have a single child and options in that child (Use div tag) 1`] = `
<div>
  <button
    className="rd-dropdown --select--"
    onBlur={[Function]}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Click me
    <span>
               
    </span>
    <i
      aria-hidden="true"
      className="fa fa-caret-down"
    />
  </button>
</div>
`;

exports[`Dropdown should have a single child and options in that child (Use div tag) 2`] = `
<div>
  <button
    className="rd-dropdown --select--"
    onBlur={[Function]}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    --select--
    <span>
               
    </span>
    <i
      aria-hidden="true"
      className="fa fa-caret-down"
    />
  </button>
</div>
`;

exports[`Dropdown should have a single child and options in that child (Use div tag) 3`] = `
<div>
  <button
    className="rd-dropdown --select--"
    onBlur={[Function]}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    --select--
    <span>
               
    </span>
    <i
      aria-hidden="true"
      className="fa fa-caret-down"
    />
  </button>
</div>
`;

exports[`Dropdown should have a single child and options in that child (Use div tag) 4`] = `
<div>
  <button
    className="rd-dropdown --select--"
    onBlur={[Function]}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    --select--
    <span>
               
    </span>
    <i
      aria-hidden="true"
      className="fa fa-caret-down"
    />
  </button>
</div>
`;
